crates/types/src/traits/block_contents.rs

```diff
@@ -88,7 +88,13 @@ pub trait BlockPayload:
     fn from_bytes(encoded_transactions: &[u8], metadata: &Self::Metadata) -> Self;
 
     /// Build the genesis payload and metadata.
-    fn genesis() -> (Self, Self::Metadata);
+    #[must_use]
+    fn genesis() -> (Self, Self::Metadata)
+    where
+        <Self as BlockPayload>::Instance: Default,
+    {
+        Self::from_transactions([], &Default::default()).unwrap()
```

Comment:
I think a default implementation makes sense, though I'm a little uncomfortable with a default implementation that may panic depending on how a non-default (`from_transactions`) is implemented.

It's not a huge deal since I guess this is just a startup panic, but is there any way to avoid this? e.g. are any of our `from_transactions` actually fallible?

Reply:
On quick review it appears that the implantation on the sequencer is used everywhere.  The only real failure case I see is [from_namespace_offsets](https://github.com/EspressoSystems/espresso-sequencer/blob/0ab566c81721352e9db3bc0e551a519dbeb8e9b0/sequencer/src/block/payload.rs#L231). I'm assuming that it isn't a huge deal b/c it is itself being unwrapped. But maybe @ggutoski,  @jbearer would have more insight.

Reply:
My opinion: it's okay because it's just a startup panic, so if there's something wrong then we'll know immediately and there's zero risk of loss. If `from_transactions` fails on an empty list then that's pretty bad and should be fixed. 😆 

Regarding the current implementation details: please keep in mind that everything in `sequencer/src/block` is currently being rewritten, so don't worry to much about what's currently in there.